03-transactions.md

```diff
@@ -120,6 +120,32 @@ If a revoked commitment transaction is published, the other party can spend this
 
     <revocation_sig> 1
 
+##### Leased channel (`option_will_fund`)
+
+If a `lease` applies to the channel, the `to_local` output of the `accepter`
+ensures the `leasor` funds are not spendable until the lease expires.
+
+In a leased channel, the `to_local` output that pays the `accepter` node
+is modified so that its CSV is equal to the greater of the
```

Comment:
We used CSV since that was already introduced to the scripts for `option_anchor_outputs`, and therefore required less surgery to the onchain scripts. :angel: 

I could further make the argument that if your peer isn't actively updating the CSV lock, then it's a good signal to go ahead and shut the channel with them but that's not exactly a water-tight argument for keeping it as a CSV ...

Reply:
> Since the lease expires at an absolute block, why do you use a CSV and not a CLTV (which would remove the need for the update_blockheight mechanism)?

I really like this suggestion!! the `update_blockheight` hack feels very hacky imo haha. But I *do* like how cleanly it reuses the CSV lock that anchors introduced. hmm

Reply:
![trade-offs](https://user-images.githubusercontent.com/31281497/129916757-04ef4e11-810c-4be6-9c37-16e49af217fb.jpg)

Reply:
Agreed, I plan on prototyping the CLTV version soon(ish). I'm not 100% sure it will work, as we're already using the `nLockTime` of commitment txs to encode data, I'll report back when it's done.

Reply:
FYI - It looks like LND [already](https://github.com/lightningnetwork/lnd/blob/master/input/script_utils.go) uses CLTV leases for Pool.  They also have CLTVs added to HTLC-success and HTLC-timeout transactions, which partially plugs the HTLC hole @ariard pointed out.

I think this is a much simpler approach.

Reply:
> Agreed, I plan on prototyping the CLTV version soon(ish). I'm not 100% sure it will work, as we're already using the nLockTime of commitment txs to encode data, I'll report back when it's done.

I don't know what I was smoking when I wrote that message, using CLTV of course works, we don't care about the `nLockTime` of the commitment tx, the `nLockTime` that matters is the one of the transaction that spends the leasor's main output.